Controller/Content/ContentController.php

```diff
@@ -107,4 +107,9 @@ protected function handleAction(string $resourceId, string $locale, string $acti
     abstract protected function handlePublish(string $resourceId, string $locale): void;
 
     abstract protected function getResourceKey(): string;
+
+    public function getLocale(Request $request)
+    {
+        return $request->query->get('locale') ?? '';
```

Comment:
`''` is a strange fallback here. this could lead to confusing behaviour. i think i would expect an error here if no locale is given

Reply:
```suggestion
        $locale = $request->query->get('locale');
        Assert::string($locale);
        
        return $locale;
```